### ISSUE TITLE
echo in loop does not print anything on Windows

### ISSUE THREAD

**Issue by cinemast** [nushell/nushell#14534]
### Describe the bug

Using echo in for loops does not work in Windows, it just does not print anything.

### How to reproduce

using this script:

```
for $x in [1 2 3] {
    echo $x
}
```

it outputs 1,2,3 in Linux, but nothing on Windows

### Expected behavior

It should output 1,2,3 on windows as well

### Configuration

```
| key                | value                                    |
| ------------------ | ---------------------------------------- |
| version            | 0.100.0                                  |
| major              | 0                                        |
| minor              | 100                                      |
| patch              | 0                                        |
| branch             |                                          |
| commit_hash        | 30f98f7e64a63a159ce70f02aa6c4c1196cd9ca7 |
| build_os           | windows-x86_64                           |
| build_target       | x86_64-pc-windows-msvc                   |
| rust_version       | rustc 1.80.1 (3f5fd8dd4 2024-08-06)      |
| rust_channel       | 1.80.1-x86_64-pc-windows-msvc            |
| cargo_version      | cargo 1.80.1 (376290515 2024-07-16)      |
| build_time         | 2024-11-13 03:41:46 +00:00               |
| build_rust_channel | release                                  |
| allocator          | mimalloc                                 |
| features           | default, sqlite, trash                   |
| installed_plugins  |                                          |
```

**Comment by fdncred** [comment]
This doesn't work in Linux or MacOS for me.
![image](https://github.com/user-attachments/assets/4825ca9f-d353-415b-a37a-fad79ff6ecb5)
I don't think it's supposed to. `echo` returns a value. If you want to print it out use `print $x`

**Comment by NotTheDr01ds** [comment]
#### tl;dr

* If you want to return values (that will then typically be printed, if they are the last thing in the pipeline), use `each` with return values
* If you want to *print* values, use `for` with `print` statements

#### Explanation

As @fdncred mentioned, the behavior here is correct - As a near-functional language, pretty much every command in Nushell returns a value.

BUT ... `for`, `while`, and `loop` are a bit of a exception to the rule (although not really). They were added as a non-functional alternative to some tasks, especially those that operate on mutable values (since they all accept a `block` rather than a `closure`.  They all return `nothing`, and there's no way to return anything else.

Also, `echo` returns a value - Its job isn't to "print" as it is in Bash.  `echo` is an *almost* useless statement, since the following two statements do *exactly* the same thing:

```nushell
echo $x   # explicitly returns the value of $x
$x            # implicitly returns the value of $x
```

**Comment by NotTheDr01ds** [comment]
> it outputs 1,2,3 in Linux, but nothing on Windows

Can you confirm again that it outputs on Linux?  It doesn't for me.  The Bash equivalent would, but not Nushell.
